File: backend/ADR_Viewer/src/main/java/com/serviceEngineering/ADR_Viewer/entity/ADR.java
package com.serviceEngineering.ADR_Viewer.entity;

import com.serviceEngineering.ADR_Viewer.div.Status;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class ADR {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    private int id;

    private String title;
    private String context;
    private String decision;
    private Status status;
    private String consequences;
    private String artifacts; //TODO: find correct type
    private String relations; //TODO: find correct type ----> probably custom type Relation

}

File: frontend/adr-viewer-ui/src/app/interfaces/adr.ts
import {Artifact} from "./artifact";
import {Relation} from "./relation";

export class ADR {
  id: number;
  title: string;
  context: string;
  decision: string;
  status: string;
  consequences: string;
  artifacts: Artifact[];
  relations: Relation[];
  importTaskId: number;
  filePath: string;
  date: string;
  commit: string;
}
